src/dynamics/video_player/player/player.js

```diff
@@ -585,7 +585,9 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
                             if (thisPlaylist.length === 1 && thisPlaylist[0].title === this.get('title')) return;
                             const showNextTime = Number(this.get("shownext")) || 0;
                             const engageTime = showNextTime + (Number(this.get("noengagenext")) || 0);
-                            if (position > showNextTime && showNextTime && !this.get("next_active")) {
+                            const duration = this.get("duration");
+                            const shouldActivateNext = duration - showNextTime >= (Number(this.get("noengagenext")) || 0);
```

Comment:
great solution @Josewd 👍 only I think will be better just in case add brackets `(duration - showNextTime)` as it can compare not with the result but only with `showNextTime`